iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -48,9 +48,14 @@ static AmqpTransportHandler()
         {
             _operationTimeout = transportSettings.OperationTimeout;
             _onDesiredStatePatchListener = onDesiredStatePatchReceivedCallback;
-            IDeviceIdentity deviceIdentity = new DeviceIdentity(connectionInfo, transportSettings, context.ProductInfo, context.ClientOptions);
+
+            // Set the context information required for setting up the connection into IotHubConnectionInfo
+            connectionInfo.AmqpTransportSettings = transportSettings;
+            connectionInfo.ProductInfo = context.ProductInfo;
+            connectionInfo.ClientOptions = context.ClientOptions;
```

Comment:
This is the only place where a DeviceIdenity instance was being created, so I modified the code to update the information into the existing IotHubConnectionInfo.